test/ex_doc/formatter/html_test.exs

```diff
@@ -592,6 +592,42 @@ defmodule ExDoc.Formatter.HTMLTest do
              ] = Jason.decode!(content)["extras"]
     end
 
+    test "custom search data is added to the sidebar", %{tmp_dir: tmp_dir} = context do
```

Comment:
```suggestion
    test "custom search data is added to the sidebar and search nodes", %{tmp_dir: tmp_dir} = context do
```

And we need to test the generated search_data.js as well. Also please add one item without anchor so we get proper coverage on those branches too.